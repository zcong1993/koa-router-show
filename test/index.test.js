
const Koa = require('koa')
const Router = require('koa-router')
const supertest = require('supertest')
const koaRouterShow = require('../')

const createRequest = (opts = {}, routerModifier = router => {
  router.get('/', () => {})
  router.get('/user/:name', () => {})
}) => {
  const app = new Koa()
  const router = new Router()
  routerModifier(router)

  koaRouterShow(router, opts)
  app.use(router.routes())
  app.use(router.allowedMethods())

  const request = supertest(app.listen())

  return request
}

test('it should add router', async () => {
  const request = createRequest()

  const res = await request
      .get('/routes')

  expect(res.status).toBe(200)
  expect(res.text).toMatchSnapshot()
})

test('it should not rewrite router', async () => {
  const request = createRequest({}, router => {
    router.get('/routes', ctx => {
      ctx.body = 'user router'
    })
  })

  const res = await request
      .get('/routes')

  expect(res.status).toBe(200)
  expect(res.text).toMatchSnapshot()
})

test('it should not add router', async () => {
  const request = createRequest({
    addRouter: false
  })

  const res = await request
      .get('/routes')

  expect(res.status).toBe(404)
})

test('it should add custom router', async () => {
  const request = createRequest({
    path: '/custom'
  })

  const res = await request
      .get('/custom')

  expect(res.status).toBe(200)
  expect(res.text).toMatchSnapshot()
})

test('it should throw error', () => {
  const router = 'bad router instance'
  expect(() => koaRouterShow(router)).toThrow('not a koa-router instance!')
})
