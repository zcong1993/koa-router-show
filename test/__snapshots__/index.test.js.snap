// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should add custom router 1`] = `
"
 Path         Methods   ParamNames 

 /            HEAD,GET             

 /user/:name  HEAD,GET  name       

"
`;

exports[`it should add router 1`] = `
"
 Path         Methods   ParamNames 

 /            HEAD,GET             

 /user/:name  HEAD,GET  name       

"
`;

exports[`it should not rewrite router 1`] = `"user router"`;
